### DISCUSSION TITLE
Read timeouts and errors from GitHub URLs

### DISCUSSION THREAD

**Discussion by briantist** [community/community#24671] [Apps, API and Webhooks]
<p>For several hours at least, have been seeing timeouts and connection errors to GitHub URLs. These are showing up mostly in my runs in GitHub actions, when trying to hit various URLs. Since I <code>pip install</code> directly from the <code>.tar.gz</code> URLs in GitHub repos, I am mostly seeing it there, but have even had failures during the setup of jobs when the runner tries to download the various GitHub Actions it needs to start the job, for example see: <a href="https://github.com/ansible-collections/community.hashi_vault/runs/4637115495?check_suite_focus=true#step:1:34" class="inline-onebox" rel="noopener nofollow ugc">Add `vault_login` lookup and module, and `vault_login_token` filter plugin (#199) · ansible-collections/community.hashi_vault@b6f4a88 · GitHub</a></p>
<p>I’ve been seeing this for a minimum of 5 hours or so, when my scheduled CI run first saw it, and I’ve been retrying runs ever since. I even hit a 500 error once just from clicking the retry button in actions.</p>
<p>GitHub status page shows no problems.</p>

**Comment by dzikoysk** [comment]
<p>Experienced similar behaviors, requests from GitHub Actions to GitHub just die, e.g.:</p>
<pre><code class="lang-auto">#1 790.1 error: RPC failed; curl 56 OpenSSL SSL_read: Connection reset by peer, errno 104
</code></pre>
<pre><code class="lang-auto">error: failed to solve: failed to read dockerfile: failed to fetch remote https://github.com/***/reposilite.git: exit status 128
</code></pre>
<p>~ <a href="https://github.com/dzikoysk/reposilite/runs/4637903635?check_suite_focus=true" class="inline-onebox" rel="noopener nofollow ugc">Enhance debug logging around proxied content · dzikoysk/reposilite@831abb3 · GitHub</a></p>

**Comment by pfeerick** [comment]
<p>Same here… we got a couple intermittent errors from the checkout step of our github action</p>
<pre><code class="lang-nohighlight">RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
</code></pre>
<p>about five hours ago, and there were extended checkout times when it did work for a couple of builds… and now it has hit hard and several runs in a row have failed completely.</p>
<p><a href="https://github.com/EdgeTX/edgetx/actions/runs/1625574976" rel="noopener nofollow ugc">https://github.com/EdgeTX/edgetx/actions/runs/1625574976</a></p>
<p>When I tried with my own repo copy, curl was dropping the connection when downloading a build dependency as well.</p>

**Comment by felixfontein** [comment]
<p>I’m also seeing this a lot in CI (GHA) in various repositories.</p>

**Comment by aminvakil** [comment]
<p>I have also faced this in the last couple of hours in github actions.</p>

**Comment by karisalonen** [comment]
<p>Same or similar problems here too.</p>

**Comment by orlangure** [comment]
<p>We are experiencing <code>go mod download</code> failures on Github Actions infrastructure:</p>
<pre><code class="lang-auto">go mod download: github.com/akeylesslabs/go-kmip@v0.2.3: invalid version: git ls-remote -q origin in /home/runner/go/pkg/mod/cache/vcs/613c275393fbb4238d0c114e32f5e44b04cbb0d0f28b4fbb0ab47dbab8a4feb8: exit status 128:
	fatal: unable to access 'https://github.com/akeylesslabs/go-kmip/': gnutls_handshake() failed: Error in the pull function.
Error: Process completed with exit code 1.
</code></pre>
<p>Packages which are not hosted on Github are loaded properly. The failures are between Github Actions Runners and Github Repositories they are trying to access.</p>

**Comment by felixfontein** [comment]
<p>I’m also seeing this a lot in repos which use AZP. In some of them, all of the recent runs already died before really starting when AZP wasn’t able to retrieve the AZP config from the repository… (See <a href="https://dev.azure.com/ansible/community.general/_build?definitionId=31" rel="noopener nofollow ugc">https://dev.azure.com/ansible/community.general/_build?definitionId=31</a> for some examples.)</p>

**Comment by briantist** [comment]
<p>Seems the problem is really a communication issue between Azure and GitHub then, given the GHA runners are also Azure VMs.</p>

**Comment by briantist** [comment, marked as answer]
<p>The incident is marked as resolved. Restarting workflows, been 5 minutes of running so far… no errors yet. :crossed_fingers:</p>

**Comment by felixfontein** [comment]
<p>I can confirm that all GHA and AZP jobs now succeed (except the ones that fail because of something else, like problems/bugs in PRs :wink: ).</p>
